examples/nft-d/nft.go

```diff
@@ -425,3 +427,59 @@ func Update(nef, manifest []byte) {
 	}
 	management.Update(nef, manifest)
 }
+
+// RoyaltyRecipient contains information about the recipient and the royalty amount.
+type RoyaltyRecipient struct {
+	Address interop.Hash160
+	Amount  int
+}
+
+// RoyaltyInfo returns a list of royalty recipients and the corresponding royalty amounts.
+func RoyaltyInfo(tokenID []byte, royaltyToken interop.Hash160, salePrice int) []RoyaltyRecipient {
+	if salePrice <= 0 {
+		panic("sale price must be positive")
+	}
+
+	callingHash := runtime.GetExecutingScriptHash()
+	if !royaltyToken.Equals(callingHash) {
+		panic("invalid royalty token")
+	}
+
+	ctx := storage.GetReadOnlyContext()
+	key := mkTokenKey(tokenID)
+	val := storage.Get(ctx, key)
+	if val == nil {
+		panic("no token found")
+	}
+	owner := val.(interop.Hash160)
+	if owner == nil {
+		panic("invalid owner")
+	}
+
+	royaltyInfoKey := append([]byte(royaltyInfoPrefix), tokenID...)
+	val = storage.Get(ctx, royaltyInfoKey)
+
+	var recipients []RoyaltyRecipient
+	if val == nil {
+		recipients = []RoyaltyRecipient{
+			{
+				Address: owner,
+				Amount:  (salePrice * 10) / 100,
+			},
+			{
+				Address: contractOwner,
+				Amount:  (salePrice * 5) / 100,
+			},
+		}
+	} else {
+		storedRecipients := val.([]any)
+		for _, r := range storedRecipients {
+			recipient := r.([]any)
+			recipients = append(recipients, RoyaltyRecipient{
+				Address: recipient[0].(interop.Hash160),
+				Amount:  recipient[1].(int),
+			})
+		}
```

Comment:
It's a dead code in fact, because you never store any royalty info in the contract. Remove this code from `nft-d`, let its royalties be fixed and evenly split between all token owners. And for `nft-nd` you may add this code that stores royalty info in the contract.

Reply:
Added `func SetRoyaltyInfo(ctx storage.Context, tokenID []byte, recipients []RoyaltyRecipient) bool ` to `nft-nd`.

Reply:
> Is it correct that I added 

No, see https://github.com/nspcc-dev/neo-go/pull/3560#discussion_r1830773733.